### ISSUE TITLE
Implement filtering option for package path for `lutaml_klass_table`

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/mn-samples-plateau#231]
Right now the command only accepts `name="NameOfClass"` for filtering, but there are many cases where duplicate names are found across the same package.

In the XMI file here, the `brid:Door` and `bldg:Door` classes have the same name but clearly their packages and what they mean are very different.

![bird](https://github.com/user-attachments/assets/de59928a-2952-4204-86e1-f9ad6fd2a042)

![bldg](https://github.com/user-attachments/assets/fad506b9-ac94-4d88-9dbf-70659b412a88)

We need to support the following additional filters:
* `package="name"` which specifies the package that contains that class directly
* `path={package:: ... ::class}` which specifies the path that matches the final element.

According to the OMG OCL 2.4, 7.5.7:
* https://www.omg.org/spec/OCL/2.4/PDF

<img width="1085" alt="Screenshot 2024-11-12 at 9 47 19 AM" src="https://github.com/user-attachments/assets/3bad324a-bd92-480e-85e8-1d3f5e6aacbb">

There is a specification for an "object path", which is:
* "Packagename::Typename"
* "Packagename::Packagename...::Typename"

We can further extend this by supporting:
* "::Packagename::Packagename::...::Typename" which is an absolute path

This is the specification.

## Expanded filter options

```adoc
== Enhanced class path specification for lutaml_klass_table

=== Purpose

The `lutaml_klass_table` command renders UML class information as tables. This enhancement adds support for OCL-style package paths to precisely identify classes across package hierarchies.

=== Path specification syntax

The command supports these path specification options:

* `name="ClassName"` - Matches a class by name (original behavior, may match multiple classes)
* `package="PackageName"` - Matches classes directly contained in the specified package
* `path="Package1::Package2::ClassName"` - Matches using relative package path
* `path="::Package1::Package2::ClassName"` - Matches using absolute package path from model root

=== Path resolution rules

==== Name-only resolution 

When only `name` is specified:

* Matches any class with the given name regardless of package location
* If multiple matches exist, the first match is used and a warning is issued

[source]
----
lutaml_klass_table::/path/to/model.xmi[name="Door"]
----

==== Package-scoped resolution

When `package` is specified:

* Matches classes directly contained in the named package
* Does not match classes in sub-packages
* Must be combined with `name`

[source]
----
lutaml_klass_table::/path/to/model.xmi[name="Door",package="Building"]
----

==== Relative path resolution

When using relative path notation:

* Path starts with package name
* Package names are separated by `::`
* Last segment is the class name
* Matches from any level in package hierarchy

[source]
----
lutaml_klass_table::/path/to/model.xmi[path="Building::Components::Door"]
----

==== Absolute path resolution 

When using absolute path notation:

* Path starts with `::`
* Package names are separated by `::`
* Last segment is the class name
* Matches only from model root

[source]
----
lutaml_klass_table::/path/to/model.xmi[path="::CityGML::Building::Components::Door"]
----

=== Edge cases

==== Package name conflicts

* If duplicate package names exist at different levels, relative paths match the first occurrence
* Use absolute paths to precisely specify location
* Warning is issued if ambiguous match is found

==== Empty/invalid paths

* Empty path segments (e.g. `A::::B`) are invalid
* Non-existent packages or classes return error
* Case sensitivity follows UML model rules

=== Examples

.Match by name only (original behavior)
[source]
----
lutaml_klass_table::/path/to/model.xmi[name="Door"]
----

.Match by package and name
[source]
----
lutaml_klass_table::/path/to/model.xmi[name="Door",package="Building"]
----

.Match using relative path
[source]
----
lutaml_klass_table::/path/to/model.xmi[path="Building::Components::Door"] 
----

.Match using absolute path
[source]
----
lutaml_klass_table::/path/to/model.xmi[path="::CityGML::Building::Components::Door"]
----

.Using custom template with absolute path
[source]
----
lutaml_klass_table::/path/to/model.xmi[path="::CityGML::Building::Door",template="/templates/building_class.liquid"]
----
```

**Comment by ReesePlews** [comment]
hi @ronaldtse thank you  for providing an issue about this enhancement discussion. 

wondering if this part of the text from above

> ==== Package-scoped resolution
> 
> When `package` is specified:
> 
> * Matches classes directly contained in the named package
> * Does not match classes in sub-packages
> * Must be combined with `name`
> 
> [source]
> 
> lutaml_klass_table::/path/to/model.xmi[name="Door",package="Building"]
> 

why is the "name" specified before the "package" in this command line, when i check the others are seem to be "package" first before "name"?

**Comment by ronaldtse** [comment]
@ReesePlews the order of these options don't matter, either way works.

**Comment by kwkwan** [comment]
Hi @ronaldtse I do not understand how relative path works?  Does it mean matching part of the absolute path?

**Comment by ReesePlews** [comment]
hi @ronaldtse that is really cool. do you mean the command line can be either:

`lutaml_klass_table::/path/to/model.xmi[name="Door",package="Building"]`

or this

`lutaml_klass_table::/path/to/model.xmi[package="Building",name="Door"]`

and the call to the statement will work ok? if so that is very flexible.

**Comment by ronaldtse** [comment]
Yes, they are identical.

**Comment by ronaldtse** [comment]
> Hi @ronaldtse I do not understand how relative path works? Does it mean matching part of the absolute path?

Yes, please refer to:
* https://github.com/metanorma/metanorma-plugin-lutaml/issues

We will discuss features there.

**Comment by ronaldtse** [comment]
@kwkwan is this task done? Thanks.

**Comment by ReesePlews** [comment]
hi @ronaldtse the issue related to this on the private repo is still open as the client is now checking all of the autogenerated uml tables. we can probably close it here but the private repo will stay open until the results are reported back. this was confirmed with @kwkwan last week.

**Comment by kwkwan** [comment]
Closed #231 as completed via https://github.com/metanorma/metanorma-plugin-lutaml/issues/161.
